table/tables/partition.go

```diff
@@ -413,6 +504,33 @@ func (t *partitionedTable) locateRangeColumnPartition(ctx sessionctx.Context, pi
 	return idx, nil
 }
 
+func (t *partitionedTable) locateListPartition(ctx sessionctx.Context, pi *model.PartitionInfo, r []types.Datum) (int, error) {
```

Comment:
This can be done in another PR, and it needs more tests

Reply:
But if I remove this, the `list partition` table can be successfully created, but DML in list partition table be wrong, it break the compatibility (⊙o⊙)…